StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/UsingCodeFixProvider.cs

```diff
@@ -0,0 +1,642 @@
+            }
+            else
+            {
+                usingsIndentation = string.Empty;
+            }
+
+            // - The strategy is to strip all using directive that are not inside a conditional directive and replace them later with a sorted list at the correct spot
+            // - The using directives that are inside a conditional directive are replaced (in sorted order) on the spot.
+            // - Conditional directives are not moved, as correctly parsing them is too tricky
+            // - No using directives will be stripped when there are multiple namespaces. In that case everything is replaced on the spot.
+            List<UsingDirectiveSyntax> stripList;
+            var replaceMap = new Dictionary<UsingDirectiveSyntax, UsingDirectiveSyntax>();
+
+            // When there are multiple namespaces, do not move using statements outside of them, only sort.
+            if (namespaceCount > 1)
+            {
+                BuildReplaceMapForNamespaces(usingsHelper, replaceMap, indentationOptions);
+                stripList = new List<UsingDirectiveSyntax>();
+            }
+            else
+            {
+                stripList = usingsHelper.GetContainedUsings(usingsHelper.RootSpan);
+            }
+
+            BuildReplaceMapForConditionalDirectives(usingsHelper, replaceMap, indentationOptions, usingsHelper.RootSpan);
+
+            var usingSyntaxRewriter = new UsingSyntaxRewriter(stripList, replaceMap);
+            var newSyntaxRoot = usingSyntaxRewriter.Visit(syntaxRoot);
+
+            if (moveInsideNamespace)
+            {
+                newSyntaxRoot = AddUsingsToNamespace(newSyntaxRoot, usingsHelper, usingsIndentation, replaceMap.Any());
+            }
+            else if (namespaceCount <= 1)
+            {
+                newSyntaxRoot = AddUsingsToCompilationRoot(newSyntaxRoot, usingsHelper, usingsIndentation, replaceMap.Any());
+            }
+
+            var newDocument = document.WithSyntaxRoot(newSyntaxRoot.WithoutFormatting());
+
+            return newDocument;
+        }
+
+        private static int CountNamespaces(SyntaxList<MemberDeclarationSyntax> members)
+        {
+            var result = 0;
+
+            foreach (var namespaceDeclaration in members.OfType<NamespaceDeclarationSyntax>())
+            {
+                result += 1 + CountNamespaces(namespaceDeclaration.Members);
+            }
+
+            return result;
+        }
+
+        private static void BuildReplaceMapForNamespaces(UsingsHelper usingsHelper, Dictionary<UsingDirectiveSyntax, UsingDirectiveSyntax> replaceMap, IndentationOptions indentationOptions)
+        {
+            var usingsPerNamespace = usingsHelper
+                .GetContainedUsings(usingsHelper.RootSpan)
+                .GroupBy(ud => ud.Parent)
+                .Select(gr => gr.ToList());
+
+            foreach (var usingList in usingsPerNamespace)
+            {
+                if (usingList.Count > 0)
+                {
```

Comment:
This didn't seem like a big win because it's not possible to do the same thing in the method that follows (`BuildReplaceMapForConditionalDirectives`).